### ISSUE TITLE
ENH:

### ISSUE THREAD

**Issue by reteps** [pandas-dev/pandas#61350]
### Feature Type

- [ ] Adding new functionality to pandas

- [x] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

Currently, the pandas [Styler](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.html) API can be used to create a HTML table from a dataframe. However, the tables it generates are not accessible: it fails [WCAG/H63](https://www.w3.org/WAI/WCAG21/Techniques/html/H63).

### Feature Description

Ensure the output generated by Styler is accessible.

- `th` with class `row_heading` needs the `row` scope

I use the current workaround to add this rule myself:

```
    html_root = lxml.html.fromstring(frame_style.to_html())
    for th in html_root.xpath("//th[contains(@class, 'row_heading')]"):
        th.set("scope", "row")
```

### Alternative Solutions

- Make the styler API more flexible for adding attributes. Currently, [set_td_classes](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.set_td_classes.html#pandas.io.formats.style.Styler.set_td_classes) and [set_table_styles](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.set_table_styles.html#pandas.io.formats.style.Styler.set_table_styles) aren't flexible enough for this, and [set_table_attributes](https://pandas.pydata.org/docs/reference/api/pandas.io.formats.style.Styler.set_table_attributes.html#pandas.io.formats.style.Styler.set_table_attributes) can't set attributes on `th` elements themselves.

**Comment by reteps** [comment]
Closing in favor of https://gitlab.com/html-validate/html-validate/-/issues/303.
